source/platform_apple.cmake

```diff
@@ -0,0 +1,48 @@
+if( NOT APPLE)
+    message(FATAL_ERROR "Don't include this on non-macOS platforms.")
+endif()
+
+include(CheckIncludeFileCXX)
+check_include_file_cxx(filesystem HAVE_FILESYSTEM)
+
+if(NOT HAVE_FILESYSTEM)
+  message(FATAL_ERROR "filesystem header NOT found. Please use LLVM Clang or upgrade to Xcode 11.")
+endif()
+
+# Darwin 19 (macOS 10.15) adds the std::filesystem::path.
+if(CMAKE_SYSTEM_VERSION VERSION_LESS 19)
```

Comment:
Can you explain what exactly does it do?

I understand the `add_compile_options` and `add_link_options` is needed. (You want to [pass the extra compiler flags](https://libcxx.llvm.org/docs/UsingLibcxx.html).) And you want to give the `-rpath` too. But I don't see, why you need to copy the `libc++*.dylib` files.

Reply:
Added explanatory paragraph. 
That is similar to what was done earlier with `libbear.dylib`. How would a user on macOS < 10.15 launch the app if their system dylib doesn't have the symbols ? So we need to ship the dylibs  ourselves. 

Or statically link to it, if possible (:

https://github.com/rizsotto/Bear/blob/2.4.4/libear/CMakeLists.txt#L32

Reply:
Thanks for adding the comments in. Now I understand it better... But more hesitant to merge it. :)

It really feels these changes are specific to one OS few versions. (Which I have no access and can't set up CI job to verify changes I later do.) I think this is a packaging problem. (I would make `brew` patch the sources before it builds the package.) I would not like to keep in upstream these changes.

- Shipping an artefact of another package seems odd to me. Static linking (even for a single library) looks more convenient.
- `libexec` and `er` targets needs no install on macOS, since the library preloading does not work on macOS. (It can be compiled (and it's desired), but do not install them.)